AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -88,122 +93,205 @@ def __init__(self):
+                "Police -> Toggle Ranking up on Police Kills",
+                lambda *args: [],
+                self.answer_toggle_police_kills_rankup
             )
         ]
 
-        self.ranks = GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, DEFAULT_RANKS)
+    # I've done the database stuff in such a way that configs are only stored if changed; otherwise the hardcoded default configs are used.
+    # Not sure how sensible this is, but it seems to work
+
+    def gsdb_get(self, plugin_state_id):
+        return GENERIC_STATE_DATABASE.arb_state.get(self.identifier, {}).get(self.plugin_state[plugin_state_id]['id'], self.plugin_state[plugin_state_id]['default'])
 
-    def ask_set_ranks(self):  # TODO Make Ctrl-C work - not sure why it isn't
+    def gsdb_set(self, plugin_state_id, data):
+        GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, {})[self.plugin_state[plugin_state_id]['id']] = data
+
+    def ask_set_ranks(self):
+        # TODO Make Ctrl-C work - not sure why it isn't --> Check Jamie's pull
+        default_text = self.gsdb_get("Ranks").copy()
+        default_text[int(self.gsdb_get("Default Rank"))] += "=default"
+        default_text.insert(0, "Add named ranks, rename old/autogenerated ranks, and set default rank."
+                               "\nThe final and penultimate ranks are reserved for Umpire and Chief of Police")
         return [
-            Label("Rename different ranks of police to fit the game theme"),
-            Label("Ranks affect position on the police page, and can be used to reward/punish/just have a bit of fun"),
-            InputWithDropDown(
-                title="Select rank to rename",
-                identifier=self.html_ids["Rank"],
-                options=[str(i[0]) + ': ' + i[1] for i in list(GENERIC_STATE_DATABASE.arb_state.get(self.identifier).items())]
+            LargeTextEntry(
+                title="Rename Ranks",
+                identifier=self.html_ids["Ranks"],
+                default="\n".join(default_text)
             ),
-            NamedSmallTextbox(
-                title="Enter new name",
-                identifier=self.html_ids["Rank Name"],
-            )
         ]
 
     def answer_set_ranks(self, htmlResponse):
-        rank = htmlResponse[self.html_ids['Rank']]
-        new_name = htmlResponse[self.html_ids['Rank Name']]
-        if new_name:
-            GENERIC_STATE_DATABASE.arb_state[self.identifier][rank[0]] = new_name
-            return [Label(f"[POLICE] Successfully renamed rank {rank} to {new_name}")]
-        else:
-            return [Label(f"[POLICE] Rename failed - no name given")]
+        answer = htmlResponse[self.html_ids['Ranks']].split("\n")
+        if answer[0] != "Add named ranks, rename old/autogenerated ranks, and set default rank." or answer[1] != "The final and penultimate ranks are reserved for Umpire and Chief of Police":
+            return [Label(f"[POLICE] Rename failed - Instructions tampered with")]
```

Comment:
You could consider starting the instructions with `#`, and deleting all lines that start with `#`. That way, you don't need to fail the user after they've given input - just leave as is if they mess with `#`. You could explain to them that lines starting with `#` are ignored (in another instr line).

```python
answer = [a for a in answer if not a.lstrip().startswith("#")]
```